cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/BasicMapReduceContext.java

```diff
@@ -81,8 +82,8 @@
   private final Map<String, LocalizeResource> resourcesToLocalize;
   private final Supplier<MapReduceTaskContextProvider> taskContextProviderSupplier;
 
-  private InputFormatProvider inputFormatProvider;
-
+  private Map<String, InputFormatProvider> inputFormatProviders;
+  private HashMap<String, Class<? extends Mapper>> mapperClasses;
```

Comment:
its not one-to-one. Often, the Mapper will be `null` for an input (if it is not explicitly set). In which case, later on in the pipeline, it will be picked up from the job.